app/frontend/components/common/asset.tsx

```diff
@@ -70,17 +70,24 @@ export const FormattedAssetItem = ({
     }
   }
 
+  const formatAssetName = (assetNameHex: string, metadata: RegisteredTokenMetadata | undefined) => {
+    if (metadata?.name) {
+      return metadata.name
+    } else if (assetName) {
+      return assetNameHex2Readable(assetNameHex)
+    }
+    return '<no name>'
+  }
+
   const FormattedAssetName = () => {
     if (type === AssetFamily.ADA) return <span className={styles.assetName}>ADA</span>
     return (
       <span>
-        {assetName ? (
-          <span className={styles.assetName}>{assetNameHex2Readable(assetName)}</span>
-        ) : (
-          <span className={styles.empty}>
-            {'<'}no name{'>'}
+        {
+          <span className={metadata || assetName ? styles.assetName : styles.empty}>
+            {formatAssetName(assetName, metadata)}
```

Comment:
Note: There is an issue with incorrect inference of type from `Map.prototype.get()` - which return type infers only to `T` instead of `T | undefined` .This is caused by disabled strict null checks. This issue is solved by: https://github.com/vacuumlabs/adalite/pull/1192